markets/perps-market/contracts/storage/PerpsPrice.sol

```diff
@@ -27,46 +28,18 @@ library PerpsPrice {
         }
     }
 
-    function getCurrentPriceData(
-        uint128 marketId
-    ) internal view returns (NodeOutput.Data memory price) {
-        Data storage self = load(marketId);
-        PerpsMarketFactory.Data storage factory = PerpsMarketFactory.load();
-        price = INodeModule(factory.oracle).process(self.feedId);
+    function getSettlementPrice(uint128 marketId) internal view returns (uint price) {
+        bytes32 nodeId = load(marketId).settleNodeId;
+        return INodeModule(PerpsMarketFactory.load().oracle).process(nodeId).price.toUint();
     }
 
     function getCurrentPrice(uint128 marketId) internal view returns (uint price) {
-        return getCurrentPriceData(marketId).price.toUint();
+        bytes32 nodeId = load(marketId).indexNodeId;
+        return INodeModule(PerpsMarketFactory.load().oracle).process(nodeId).price.toUint();
```

Comment:
One though about this new pattern.. So this can revert if the price is too stale, right?
This price function is used by many views, example: `getMarketSummary`. 
So does this means that a frontend calling `getMarketSummary` might not get any result until them or someone else request/ pay for a price update?